parse/IntComplexValueRefParser.cpp

```diff
@@ -102,6 +102,30 @@ namespace parse {
                     ) [ _val = new_<ValueRef::ComplexVariable<int> >(_a, _b, _c, _f, _d, _e) ]
                 ;
 
+            ship_parts_owned
+                =   (
+                            tok.ShipPartsOwned_ [ _a = construct<std::string>(_1) ]
+                        >-( parse::label(Empire_token) >    int_value_ref [ _b = _1 ] )
+                        >-( parse::label(Name_token)   >    string_value_ref [ _d = _1 ] )
+                    ) [ _val = new_<ValueRef::ComplexVariable<int>>(_a, _b, _c, _f, _d, _e) ]
+                ;
+
+            ship_part_class_owned
+                = (
+                            tok.ShipPartClassOwned_ [ _a = construct<std::string>(_1) ]
+                        >-( parse::label(Empire_token) >    int_value_ref [ _b = _1 ] )
+                        >-( parse::label(Class_token)  >
+                            ( tok.ShortRange_       | tok.FighterBay_   | tok.FighterWeapon_
+                            | tok.Shield_           | tok.Armour_
+                            | tok.Troops_           | tok.Detection_    | tok.Stealth_
+                            | tok.Fuel_             | tok.Colony_       | tok.Speed_
+                            | tok.General_          | tok.Bombard_      | tok.Research_
+                            | tok.Industry_         | tok.Trade_        | tok.ProductionLocation_
+                            ) [ _d = new_<ValueRef::Constant<std::string> >(_1) ]
```

Comment:
should probably be consistent about whether to use >> or > > when closing nested templates